### PR TITLE
 use unified HCO conditions

### PR DESCRIPTION
**Pull request by phoracek**
Following PR will finish the integration by introducing Upgradable condition.

### CODE REVIEW

File: pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package v1alpha1
 
 import (
+	conditions "github.com/djzager/custom-resource-status/conditions/v1"
```
Comment (qinqon) [inline pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go]: conditionsv1

File: pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -5,6 +5,7 @@
 package v1alpha1
 
 import (
+	v1 "github.com/djzager/custom-resource-status/conditions/v1"
```
Comment (qinqon) [inline pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go]: conditionsv1

Comment (qinqon) [inline pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go]: conditionsv1

File: test/check/check.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ import (
 	"strings"
 	"time"
 
+	conditions "github.com/djzager/custom-resource-status/conditions/v1"
```
Comment (qinqon) [inline test/check/check.go]: conditionsv1

File: test/check/conditions.go [hunk 4]
```diff
@@ -1,19 +1,18 @@
 package check
 
 import (
+	conditions "github.com/djzager/custom-resource-status/conditions/v1"
```
Comment (qinqon) [inline test/check/conditions.go]: conditionsv1

**Comment by phoracek** [comment]
ci test please

**Comment by phoracek** [comment]
/hold

lifecycle tests fail

File: pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package v1alpha1
 
 import (
+	conditionsv1 "github.com/djzager/custom-resource-status/conditions/v1"
```
[the remark this answers is not in the document]
Comment (phoracek) [reply pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go]: It should be eventually moved under openshift, but for now it lives in @djzager 's repo

File: pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go (hunk 2 above)
Comment (phoracek) [reply pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go]: We can't

File: pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (djzager) [reply pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go]: https://github.com/openshift/custom-resource-status/pull/1

File: pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go [hunk 6]
```diff
@@ -5,6 +5,7 @@
 package v1alpha1
 
 import (
+	v1 "github.com/openshift/custom-resource-status/conditions/v1"
```
Comment (qinqon) [inline pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go]: conditionsv1

Comment (phoracek) [reply pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go]: As Sebastian commented, it is generated and we have no control over the name.

**Comment by phoracek** [comment]
ci test please

File: pkg/controller/statusmanager/status_manager.go [hunk 7]
```diff
@@ -103,11 +104,11 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...opv1a
 
 	// In case that the status field has been updated with "Progressing" condition, make sure that
 	// "Ready" condition is set to False, even when not explicitly set.
-	progressingCondition := getCondition(&config.Status, opv1alpha1.NetworkAddonsConditionProgressing)
+	progressingCondition := conditionsv1.FindStatusCondition(config.Status.Conditions, conditionsv1.ConditionProgressing)
```
Comment (djzager) [inline pkg/controller/statusmanager/status_manager.go]: you could simplify this to `conditionsv1.IsConditionTrue(config.Status.Conditions, conditionsv1.ConditionProgressing)`.

Comment (phoracek) [reply pkg/controller/statusmanager/status_manager.go]: Thanks David, will fix it. I think we are still having issues during upgrade. Plus I will need to implement Upgradable in a following PR, but should not be much hassle.

**Comment by phoracek** [comment]
Conditions check doesn't work since we broke compatibility with conditions from previous versions, gotta add compatibility mode or disable for older versions.

**Comment by phoracek** [comment]
Will create a new API version `v1alpha2`.

**Comment by phoracek** [comment]
@SchSeba @qinqon tests are passing, I think we are ready for merge. Could you take another look please?

File: pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go (hunk 6 above)
Comment (qinqon) [reply pkg/apis/networkaddonsoperator/v1alpha1/zz_generated.deepcopy.go]: ack

**Comment by phoracek** [comment]
@SchSeba nice catch. Fixed.

Only README has been changed since last time CI passed. Merging.

**Comment by phoracek** [comment]
Now we have to add Upgradable condition.

**Pull request merged**